pkg/pillar/utils/file/file.go

```diff
@@ -14,11 +14,45 @@ import (
 	"time"
 
 	"github.com/lf-edge/eve/pkg/pillar/base"
+	"github.com/sirupsen/logrus"
 	"golang.org/x/sys/unix"
 )
 
 const maxCounterReadSize = 16384 // Max size of counter file
 
+func dirSync(dirName string) error {
+	f, err := os.OpenFile(dirName, os.O_RDONLY, 0755)
+	if err != nil {
+		return err
+	}
+
+	err = f.Sync()
+	if err != nil {
+		return err
```

Comment:
You need to Close the file, either using the defer pattern or doing it in explicitly in this error path.

Reply:
indeed. Forgot that It's not c++